### PR TITLE
Policy regression tests

### PR DESCRIPTION
**Pull request by kedz**
**Proposed changes**:
- Add "story_report.json" task mapping to  .github/scripts/mr_generate_summary.py and  .github/scripts/mr_publish_results.py so that policy action prediction results can get logged.
- Added a CI policy regression test that would look for the label: status:policy-regression-tests, similar to how status: model-regression-tests works for nlu models.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for opening a draft pull request 🚀If you have any questions, you can direct them to @Ghostvv ✨

File: .github/workflows/ci-policy-regression.yml [hunk 1]
```diff
@@ -0,0 +1,632 @@
+# The docs: https://www.notion.so/rasa/The-CI-for-model-regression-tests-92af7185e08e4fb2a0c764770a8e9095
```
Comment (tczekajlo) [inline .github/workflows/ci-policy-regression.yml:1]: I think we don't need a new workflow for it, we can add an additional condition to `if`s in the current model regression tests CI and it should be enough (look comments below)

File: .github/workflows/ci-policy-regression.yml [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+    steps:
+      - name: Find and cancel old builds of this branch
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: "${{ secrets.GITHUB_TOKEN }}"
+
+  read_test_configuration:
+    name: Reads tests configuration
+    needs: cleanup_runs
+    if: "github.repository == 'RasaHQ/rasa' && contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests')"
```
Comment (tczekajlo) [inline .github/workflows/ci-policy-regression.yml:32]:
```suggestion
    if: "github.repository == 'RasaHQ/rasa' && (contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests') || contains(github.event.pull_request.labels.*.name, 'status:model-regression-tests'))"
```
```suggestion
    if: "github.repository == 'RasaHQ/rasa' && contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests')"
```

File: .github/workflows/ci-policy-regression.yml [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+          The model regression tests have started. It might take a while, please be patient.
+          As soon as results are ready you'll see a new comment with the results.
+
+          Used configuration can be found in [the comment.](https://github.com/${{ github.repository }}/pull/${{ github.event.number}}#issuecomment-${{ steps.fc_config_update.outputs.comment-id }})
+
+  deploy_runner_gpu:
+    name: Deploy Github Runner - GPU
+    needs: read_test_configuration
+    runs-on: ubuntu-latest
+    if: "contains(github.event.pull_request.labels.*.name, 'runner:gpu') && github.repository == 'RasaHQ/rasa' && contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests') && needs.read_test_configuration.outputs.configuration_id != ''"
```
Comment (tczekajlo) [inline .github/workflows/ci-policy-regression.yml:141]:
```suggestion
    if: "contains(github.event.pull_request.labels.*.name, 'runner:gpu') && github.repository == 'RasaHQ/rasa' && (contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests') || contains(github.event.pull_request.labels.*.name, 'status:model-regression-tests')) && needs.read_test_configuration.outputs.configuration_id != ''"
```
```suggestion
    if: "contains(github.event.pull_request.labels.*.name, 'runner:gpu') && github.repository == 'RasaHQ/rasa' && contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests') && needs.read_test_configuration.outputs.configuration_id != ''"
```

File: .github/workflows/ci-policy-regression.yml [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+    - read_test_configuration
+    env:
+      # Determine where CUDA and Nvidia libraries are located. TensorFlow looks for libraries in the given paths
+      LD_LIBRARY_PATH: "/usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64"
+      ACCELERATOR_TYPE: "GPU"
+    runs-on: [self-hosted, gpu]
+    strategy:
+      max-parallel: 1
+      matrix: ${{fromJson(needs.read_test_configuration.outputs.matrix)}}
+    if: "contains(github.event.pull_request.labels.*.name, 'runner:gpu') && github.repository == 'RasaHQ/rasa' && contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests') && needs.read_test_configuration.outputs.configuration_id != ''"
```
Comment (tczekajlo) [inline .github/workflows/ci-policy-regression.yml:186]:
```suggestion
    if: "contains(github.event.pull_request.labels.*.name, 'runner:gpu') && github.repository == 'RasaHQ/rasa' && (contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests') || contains(github.event.pull_request.labels.*.name, 'status:model-regression-tests')) && needs.read_test_configuration.outputs.configuration_id != ''"
```
```suggestion
    if: "contains(github.event.pull_request.labels.*.name, 'runner:gpu') && github.repository == 'RasaHQ/rasa' && contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests') && needs.read_test_configuration.outputs.configuration_id != ''"
```

File: .github/workflows/ci-policy-regression.yml [hunk 5]
```diff
@@ -0,0 +1,632 @@
...
+          poetry run python -m spacy link --force de_core_news_md de
+
+      - name: Validate that GPUs are working
+        if: steps.set_dataset_config_vars.outputs.is_dataset_exists == 'true' && steps.set_dataset_config_vars.outputs.is_config_exists == 'true'
+        run: |-
+          poetry run python -c 'from tensorflow.python.client import device_lib; print(device_lib.list_local_devices())' || true
+
+      - name: Run test
+        id: run_test
+        if: steps.set_dataset_config_vars.outputs.is_dataset_exists == 'true' && steps.set_dataset_config_vars.outputs.is_config_exists == 'true'
```
Comment (tczekajlo) [inline .github/workflows/ci-policy-regression.yml:264]:
```suggestion
        if: steps.set_dataset_config_vars.outputs.is_dataset_exists == 'true' && steps.set_dataset_config_vars.outputs.is_config_exists == 'true' && contains(github.event.pull_request.labels.*.name, 'status:policy-regression-tests')
```
```suggestion
        if: steps.set_dataset_config_vars.outputs.is_dataset_exists == 'true' && steps.set_dataset_config_vars.outputs.is_config_exists == 'true'
```

And here (to the CI) we can add a step that runs on the `status:model-regression-tests` label

File: .github/workflows/ci-policy-regression.yml [hunk 6]
```diff
@@ -0,0 +1,632 @@
...
+          cat $SUMMARY_FILE
+
+      - name: Upload an artifact with the report
+        if: steps.set_dataset_config_vars.outputs.is_dataset_exists == 'true' && steps.set_dataset_config_vars.outputs.is_config_exists == 'true'
+        uses: actions/upload-artifact@v2
+        with:
+          name: report.json
+          path: ./report.json
+
+  model_regression_test_cpu:
```
Comment (tczekajlo) [inline .github/workflows/ci-policy-regression.yml:317]: For this job, you have to do the same changes as for the `model_regression_test_gpu`

**Review by tczekajlo — COMMENTED** [review]
I've left several comments.

Additionally, you have to modify the [model_regression_test_results.tmpl](https://github.com/RasaHQ/rasa/blob/main/.github/templates/model_regression_test_results.tmpl) template if you wanna publish results as a PR comment.

**Comment by HotThoughts** [comment]
Hey @kedz, is the PR still relevant to the policy regression test? Did https://github.com/RasaHQ/rasa/pull/8338 cover it?

**Comment by kedz** [comment]
@HotThoughts Yes this is a defunct branch and #8338 did indeed address these things. I'll close it out.

**Pull request closed without merging**